ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionOrderMapper.java

```diff
@@ -0,0 +1,132 @@
+        if (ccdDirectionOrder == null || directionOrderData == null) {
+            return null;
+        }
+
+        DirectionOrder.DirectionOrderBuilder builder = DirectionOrder.builder();
+
+        Optional.ofNullable(directionOrderData.getEstimatedHearingDuration()).ifPresent(estimatedDuration ->
+            builder.estimatedHearingDuration(HearingDurationType.valueOf(estimatedDuration.name())));
+
+        Optional.ofNullable(directionOrderData.getHearingCourt()).ifPresent(hearingCourt ->
+            builder.hearingCourt(HearingCourtType.valueOf(hearingCourt.name())));
+
+        Optional.ofNullable(directionOrderData.getPaperDetermination()).ifPresent(paperDetermination ->
+            builder.paperDetermination(YesNoOption.valueOf(paperDetermination.name())));
+
+        DirectionOrder directionOrder = builder
+            .createdOn(ccdDirectionOrder.getCreatedOn())
+            .hearingCourtAddress(addressMapper.from(ccdDirectionOrder.getHearingCourtAddress()))
+            .preferredDQCourt(directionOrderData.getPreferredDQCourt())
+            .preferredCourtObjectingReason(directionOrderData.getPreferredCourtObjectingReason())
+            .newRequestedCourt(directionOrderData.getNewRequestedCourt())
+            .extraDocUploadList(directionOrderData.getExtraDocUploadList().stream()
+                .map(CCDCollectionElement::getValue)
+                .collect(Collectors.toList()))
+            .build();
+
+        addUploadDocumentDirection(directionOrderData, directionOrder);
+        addEyeWitnessDirection(directionOrderData, directionOrder);
+
+        asStream(directionOrderData.getOtherDirections())
+            .map(CCDCollectionElement::getValue)
+            .forEach(ccdOrderDirection -> directionOrder.addDirection(Direction.builder()
+                .directionType(DirectionType.valueOf(ccdOrderDirection.getExtraOrderDirection().name()))
+                .directionComment(ccdOrderDirection.getDirectionComment())
+                .directionParty(DirectionParty.valueOf(ccdOrderDirection.getForParty().name()))
+                .directionActionedBy(ccdOrderDirection.getSendBy())
+                .directionHeaderType(getDirectionHeaderType(ccdOrderDirection))
+                .expertReports(asStream(ccdOrderDirection.getExpertReports())
+                    .filter(Objects::nonNull)
+                    .map(CCDCollectionElement::getValue)
+                    .collect(Collectors.toList()))
+                .extraDocuments(asStream(ccdOrderDirection.getExtraDocUploadList())
+                    .map(CCDCollectionElement::getValue)
+                    .collect(Collectors.toList()))
+                .build()));
+
+        return directionOrder;
+    }
+
+    private DirectionHeaderType getDirectionHeaderType(CCDOrderDirection ccdOrderDirection) {
+        return ccdOrderDirection.getOtherDirectionHeaders() == null
```

Comment:
I prefer `Optional.ofNullable(...).map()` in these case, but it's aesthetics

Reply:
done